advanced/kernel-messaging/README.md

```diff
@@ -10,85 +10,89 @@
 ## Custom Kernel Interactions: Kernel Managment and Messaging
 
 One of the main features of JupyterLab is the possibility to manage and
-interact underlying compute kernels. In this section, we explore how to
-start a kernel and execute a simple command on it.
+interact with execution [kernels](https://jupyter-client.readthedocs.io/en/latest/kernels.html). In this example, youu will explore how to
+start a kernel and execute a command on it.
 
 ## Component Overview
 
-In terms of organization of this app, we want to have these components:
+This example is structured in four files:
 
-- `index.ts`: a JupyterLabPlugin that initializes the plugin and registers commands, menu tabs and a launcher
-- `panel.ts`: a panel class that is responsible to initialize and hold the kernel session, widgets and models
+- `index.ts`: the JupyterLab frontend plugin that initializes the plugin and registers commands, add a menu entry and a launcher item,
+- `panel.ts`: a panel class that is responsible to initialize and hold the kernel session, widgets and model
 - `model.ts`: a KernelModel class that is responsible to execute code on the kernel and to store the execution result
 - `widget.tsx`: a KernelView class that is responsible to provide visual elements that trigger the kernel model and display its results
 
-The `KernelView` displays the `KernelModel` with some react html elements and
-needs to get updated when `KernelModel` changes state, i.e. retrieves a new
-execution result. Jupyterlab provides a two class model for such classes,
-a `VDomRendered` that has a link to a `VDomModel` and a `render` function.
-The `VDomRendered` listens to a `stateChanged` signal that is defined by the
-`VDomModel`. Whenever the `stateChanged` signal is emitted, the
-`VDomRendered` calls its `render` function again and updates the html elements
-according to the new state of the model.
+The `KernelView` displays the `KernelModel` thanks to some React HTML elements and
+get updated when the `KernelModel` state changes, i.e. retrieves a new
+execution result.
 
 ## Initializing and managing a kernel session (`panel.ts`)
 
 Jupyterlab provides a class `ClientSession`
-([documentation](http://JupyterLab.github.io/JupyterLab/classes/_apputils_src_clientsession_.clientsession.html))
-that manages a single kernel session. Here are the lines that we need to start
-a kernel with it:
+([see the documentation](https://jupyterlab.github.io/jupyterlab/apputils/classes/clientsession.html))
+that manages a single kernel session. The construction and initialization
+of such session is done like this:
```

Comment:
Or maybe just: "To initialize a `ClientSession`:" ?

Reply:
Yep, shorter is simpler.